src/components/Warehouse/WarehouseDetailsContainer.js

```diff
@@ -0,0 +1,85 @@
+import editIcon from "../../assets/icons/edit-white-24px.svg";
+import "./WarehouseDetailsContainer.scss";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import BackIcon from "../../assets/icons/arrow_back-24px.svg";
+import { useParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+
+const WarehouseDetailsContainer = () => {
+  const navigate = useNavigate();
+  const [warehouseData, setWarehouseData] = useState([]);
+  const { id } = useParams();
+
+  // This useEffect fetches the Warehouse data when the component mounts
+  useEffect(() => {
+    const fetchWarehouse = async () => {
+      try {
+        const resp = await axios.get(
+          `http://localhost:8080/api/warehouses/${id}`
```

Comment:
As a good practice try having these port numbers into a separate env file. So that if port number changes you dont need to change it in multiple places